File: fsharp/cars-assemble/CarsAssemble.fs
module CarsAssemble

let successRate (speed: int): float =
    if speed = 0 then
        0.0
    elif speed >= 1 && speed < 5 then
        1.0
    elif speed >= 5 && speed < 9 then
        0.9
    elif speed = 9 then
        0.80
    elif speed = 10 then
        0.77
    else
        failwith "Speed should be 0 to 10 inclusive"

let productionRatePerHour (speed: int): float =
    let carsProducedPerHour = 221
    float speed * float carsProducedPerHour * successRate speed

let workingItemsPerMinute (speed: int): int =
    int (productionRatePerHour speed / 60.0)

File: java/cars-assemble/src/main/java/CarsAssemble.java
public class CarsAssemble {

    private static final int CARS_PER_HOURS = 221;

    private double successRate(int speed) {
        if (speed == 0) {
            return 0;
        } else if (speed <= 4) {
            return 1.0;
        } else if (speed <= 8) {
            return 0.9;
        } else if (speed == 9) {
            return 0.8;
        } else {
            return 0.77;
        }
    }

    public double productionRatePerHour(int speed) {
        return speed * CARS_PER_HOURS * successRate(speed);
    }

    public int workingItemsPerMinute(int speed) {
        return (int) Math.floor(productionRatePerHour(speed) / 60);
    }
}
